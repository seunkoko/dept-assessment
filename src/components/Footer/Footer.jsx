// eslint-disable-next-line
import React from 'react';
import { Link } from 'react-router-dom';

import './Footer.scss';

import fblogo from '../../assets/images/logos/icon - fb.svg';
import twlogo from '../../assets/images/logos/icon - twitter.svg';
import inlogo from '../../assets/images/logos/icon - instagram.svg';


const Footer = (props) => {
	return (
    <>
      <footer className="page-footer footer-style">
        <div className="main-footer">
          <div className="footer-menu">
            <div className="footer-first-col hide-on-mobile">
              <p>
                DEPT
              </p>    
            </div>

            <div className="footer-second-col">
              <p><a href="#werk">WORK</a></p>

              <p><a href="#deinsten">SERVICE</a></p>

              <p><a href="#stories">STORIES</a></p>

              <p><a href="#partners">ABOUT</a></p>

              <p><a href="#vacatures">CAREERS</a></p>

              <p><a href="#contact">CONTACT</a></p>
            </div>

            <div className="footer-third-col">
              <img src={fblogo} alt="fb-logo" />
              <img src={twlogo} alt="tw-logo" />
              <img src={inlogo} alt="in-logo" />
            </div>
          </div>

          <div className="footer-copyright">
            <div className="footer-first-col hide-on-phone"></div>

            <div className="footer-second-col">
              <p>
                Chamber of Commerce: 63464101 
              </p>

              <p>
                VAT: NL 8552.47.502.B01 
              </p>

              <p>
                Terms and conditions
              </p>
            </div>

            <div className="footer-third-col">
              <p>
                Â© 2018 Dept Agency
              </p>
            </div>
          </div>

        </div>

        <div className="to-top hide-on-phone">
          <div id="return-to-top">
            <Link to="#">
              <i className="fas fa-long-arrow-alt-up"/>
              <p>
                TOP
              </p>
            </Link>
          </div>
        </div>
      </footer>
    </>
	);
};

export default Footer;
