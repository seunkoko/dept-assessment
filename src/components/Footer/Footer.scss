@import "../../stylesheets/variables";

.footer-style {
  position: absolute;
  left: 0;
  right: 0;
  height: 300px;
  width: 100%;
  display: grid;
  grid-template-columns: 90% 10%;

  @media (max-width: $phone-width) {
    display: flex;
    height: auto;
	}
}

.footer-first-col {
  font-size: $xlarge-font;
}

.footer-second-col {

  > p {
    display: inline-block;
    padding: 20px 15px 0 0;

    @media (max-width: $phone-width) {
      display: flex;
      padding: 0;
    }
  }
}

.main-footer {
  background-color: $footer-black;
  width: 100%;
  display: grid;
  grid-template-rows: auto auto;
  letter-spacing: 1px;
  font-family: 'Teko-Regular';

  > div {
    display: grid;
    grid-template-columns: 15% 65% 20%;

    @media (max-width: $mobile-width) {
      display: grid;
      grid-template-columns: 70% 30%;
    }

    @media (max-width: $phone-width) {
      display: flex;
    }
  }

  > .footer-menu {
    color: $primary-white;
    width: 80%;
    height: 40%;
    margin: auto auto 0 auto;
    border-bottom: 1px solid $footer-border-gray;

    > .footer-second-col {

      > p {
        cursor: pointer;

        > a {
          color: $primary-white;

          &:hover {
            text-decoration: none;
            color: $primary-blue;
          }
        }

        &:hover {
          color: $primary-blue;
        }
      }
    }

    > .footer-third-col {
      text-align: right;

      > img {
        padding: 20px 15px 0 0;
        cursor: pointer;

        @media (max-width: $phone-width) {
          padding: 0 0 30px 0;
        }
      }

      > img:last-child {
        padding-right: 0;
      }

      @media (max-width: $phone-width) {
        display: flex;
        flex-direction: column;
      }
    }

    @media (max-width: $laptop-width) {
      height: auto;
    }

    @media (max-width: $phone-width) {
      justify-content: space-between;
      padding-bottom: 30px;
      font-size: 24px;
    }
  }

  > .footer-copyright {
    width: 80%;
    height: 40%;
    margin: 0 auto auto auto;
    font-size: $small-font;

    > div {

      > p {
        color: $footer-gray;

        @media (max-width: $phone-width) {
          margin: 0;
          padding: 0;
        }
      }
    }
    
    > .footer-second-col {

      > p {
        padding: 40px 25px 0 0;

        @media (max-width: $laptop-width) {
          margin: 0;
          padding: 0 25px 0 0;
        }
      }
    }

    > .footer-third-col {
      text-align: right;

      > p {
        padding: 30px 0 0 0;

        @media (max-width: $laptop-width) {
          padding: 0;
        }
      }

      @media (max-width: $laptop-width) {
        text-align: left;
      }

      @media (max-width: $phone-width) {
        margin-top: 20px;
      }
    }

    @media (max-width: $laptop-width) {
      display: flex;
      flex-direction: column;
      padding-top: 30px;
      height: auto;
    }
  }

  @media (max-width: $phone-width) {
    display: flex;
    flex-direction: column;
    padding: 50px 0;
    
    > .footer-menu {
      margin: 0 auto;
      width: 90%;
    }

    > .footer-copyright {
      margin: 0 auto;
      width: 90%;
    }
	}
}

.to-top {
  font-size: 24px;
  font-family: 'Teko-Regular';
  background-color: $primary-white;
  text-align: center;

  > div {
    display: inline-block;
    margin-top: 100px;
    max-height: 90px;
    max-width: 50px;
    padding: 10px;
    border-radius: 25px;

    > a {
      color: $primary-blue;
  
      &:hover {
        text-decoration: none;
      }
    }

    &:hover {
      border: 1px solid $primary-blue;
      cursor: pointer;
    }
  }
}
