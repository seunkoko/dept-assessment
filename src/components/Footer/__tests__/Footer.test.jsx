import React from 'react';

import { shallow } from 'enzyme';

import Footer from '../Footer';

describe('Footer', () => {
	const component = shallow(<Footer />);

	it('should render Footer component', () => {
		expect(component).toMatchSnapshot();
  });
  
});