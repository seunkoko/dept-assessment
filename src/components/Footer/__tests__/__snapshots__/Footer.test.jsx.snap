// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render Footer component 1`] = `
<Fragment>
  <footer
    className="page-footer footer-style"
  >
    <div
      className="main-footer"
    >
      <div
        className="footer-menu"
      >
        <div
          className="footer-first-col hide-on-mobile"
        >
          <p>
            DEPT
          </p>
        </div>
        <div
          className="footer-second-col"
        >
          <p>
            WORK
          </p>
          <p>
            SERVICE
          </p>
          <p>
            STORIES
          </p>
          <p>
            ABOUT
          </p>
          <p>
            CAREERS
          </p>
          <p>
            CONTACT
          </p>
        </div>
        <div
          className="footer-third-col"
        >
          <img
            alt="fb-logo"
            src="test-file-stub"
          />
          <img
            alt="tw-logo"
            src="test-file-stub"
          />
          <img
            alt="in-logo"
            src="test-file-stub"
          />
        </div>
      </div>
      <div
        className="footer-copyright"
      >
        <div
          className="footer-first-col hide-on-phone"
        />
        <div
          className="footer-second-col"
        >
          <p>
            Chamber of Commerce: 63464101
          </p>
          <p>
            VAT: NL 8552.47.502.B01
          </p>
          <p>
            Terms and conditions
          </p>
        </div>
        <div
          className="footer-third-col"
        >
          <p>
            Â© 2018 Dept Agency
          </p>
        </div>
      </div>
    </div>
    <div
      className="to-top hide-on-phone"
    >
      <div
        id="return-to-top"
      >
        <Link
          to="#"
        >
          <i
            className="fas fa-long-arrow-alt-up"
          />
          <p>
            TOP
          </p>
        </Link>
      </div>
    </div>
  </footer>
</Fragment>
`;
